src/pages/Swap/LimitOrderBox/components/LimitOrderForm/LimitOrderForm.tsx

```diff
@@ -391,6 +391,50 @@ export function LimitOrderForm({ account, provider, chainId }: LimitOrderFormPro
     })
   }
 
+  const [limitOrderPrices, setLimitOrderPrices] = useState<{ buy: number; sell: number }>({ buy: 0, sell: 0 })
+
+  const getMarketPriceDiff = useCallback(async () => {
+    const signer = provider.getSigner()
+    if (limitOrder.buyToken && limitOrder.sellToken) {
+      const order = JSON.parse(JSON.stringify(limitOrder))
+
+      const token = limitOrder.kind === LimitOrderKind.SELL ? sellTokenAmount : buyTokenAmount
+
+      order.sellAmount = parseUnits('10', token.currency.decimals).toString()
+
+      const cowQuote = await getQuote({
+        chainId,
+        signer,
+        order: { ...order, expiresAt: dayjs().add(expiresIn, expiresInUnit).unix() },
+      })
+
+      if (cowQuote) {
+        const {
+          quote: { buyAmount, sellAmount },
+        } = cowQuote
+
+        if (limitOrder.kind === LimitOrderKind.SELL)
+          setLimitOrderPrices(limitPrice => ({
+            ...limitPrice,
+            buy: parseFloat(formatUnits(buyAmount, buyTokenAmount.currency.decimals) ?? 0) / 10,
```

Comment:
this code is very similar to the one bellow, maybe an opportunity to make a function?